src/db/models/publication/manager.rs

```diff
@@ -128,6 +128,28 @@ impl super::TxManager for DatabaseTransaction {
         Ok(row)
     }
 
+    /// Filter publications by `name` and `stele_id` which is not revoked.
+    ///
+    /// # Errors
+    /// Errors if can't establish a connection to the database.
+    async fn find_first_by_date_and_stele_non_revoked(
+        &mut self,
+        date: &str,
+        stele: &str,
+    ) -> anyhow::Result<Publication> {
+        let statement = "
+            SELECT *
+            FROM publication
+            WHERE date = $1 AND stele = $2 AND revoked = 0
+        ";
+        let row = sqlx::query_as::<_, Publication>(statement)
+            .bind(date)
+            .bind(stele)
+            .fetch_one(&mut *self.tx)
+            .await?;
+        Ok(row)
+    }
```

Comment:
Just to understand this: Is this doing sort by publication name somehow, or how does it ensure it will get the correct row from publication table? Or is this revoked flag set to true on all old publications that are on the same date, so the only one that is not revoked is then latest one?

Reply:
Yup, revoked flag is set to true for older publications, so this correctly gets the latest (non-revoked publication)

Reply:
Awesome, thanks!